### ISSUE TITLE
v3 issues with interaction

### ISSUE THREAD

**Issue by nt-pch** [pixijs/pixi.js#1901]
Hi folks, first... thanks for such a great library! 

I'm noticing differences in how v3 works in comparison to v2. I'm running Chrome 43.0.2357.124 on a Max OS 10.10.3.

I have a graphic red box in a container. The red box is listening to mouse events. http://jsfiddle.net/ar4pjetx/

1) If there is another non-interactive graphic on top of the red box and the red box's parent container has set ‘interactive = true', the red box no longer responds to events where the green box is overlapping, i.e., when mousing over both boxes, if the green box is present, then the red box thinks that the mouse is no longer over it. My expectation is that the box events should not be blocked by a non-interactive object on top of it. This is how v2 worked.
- If I remove line 32 (stage.interactive = true), then the events fire as expected.

2) When logging events in Chrome, it shows an object with ‘target: Graphics’, and logging the object’s given name shows ‘bg’. But on opening the event object, it shows an object with ‘target: Container’ and a name of ‘stage’ (the parent container). I have each event logging two things, the name, and the object. Open the object.target and you can see the name is different inside.
- If I remove line 32 (stage.interactive = true), then the logging of objects shows up as expected. Both the object overview and the individual items in the object show both ‘target: Graphics’ and ‘name: “bg”’.

3) The red box’s ‘over’ event fires the first time this fiddle is run even though the mouse is not over the red box. 

4) The red box’s ‘out’ event fires the first time the mouse moves into the canvas area even though the mouse was never over the red box. (This is most likely because the ‘over’ event was incorrectly fired so it probably thinks the mouse is over the red box to begin with.)

5) Right clicking causes the ‘over’ and ‘out’ events to fire when the right click content box shows up. Expectation: is that the context box is not part of the canvas and should not trigger events.

**Comment by endel** [comment]
Also related with https://github.com/GoodBoyDigital/pixi.js/issues/1837 and https://github.com/GoodBoyDigital/pixi.js/issues/1884

There is a patch/pull-request for this here: https://github.com/GoodBoyDigital/pixi.js/pull/1894

**Comment by stephenweaver** [comment]
added a comment to the PR https://github.com/pixijs/pixi.js/pull/1894
this issues is not resolved with that fix. I think if we can figure out why targetTouches includes all touches on the canvas rather than just the pixi object, the events will get sent to the correct objects. maybe?

**Comment by englercj** [comment]
Is this issue still active on latest v4 `dev`?

**Comment by hubletkoresh** [comment]
Definitely still there, and it seems worse: because the events now appear to fire regardless of whether I am actually hovering over the element or not: [http://jsfiddle.net/ar4pjetx/15/](http://jsfiddle.net/ar4pjetx/15/)

**Comment by staff0rd** [comment]
@hubletkoresh that particular functionality can be switched with;

~~~
renderer.plugins.interaction.moveWhenInside = true;
~~~

> If there is another non-interactive graphic on top of the red box and the red box's parent container has set ‘interactive = true', the red box no longer responds to events where the green box is overlapping, i.e., when mousing over both boxes, if the green box is present, then the red box thinks that the mouse is no longer over it. My expectation is that the box events should not be blocked by a non-interactive object on top of it. This is how v2 worked.

>If I remove line 32 (stage.interactive = true), then the events fire as expected.

Still occurs in v4

>When logging events in Chrome, it shows an object with ‘target: Graphics’, and logging the object’s given name shows ‘bg’. But on opening the event object, it shows an object with ‘target: Container’ and a name of ‘stage’ (the parent container). I have each event logging two things, the name, and the object. Open the object.target and you can see the name is different inside.

No longer occurs

> The red box’s ‘over’ event fires the first time this fiddle is run even though the mouse is not over the red box.

No longer occurs

> The red box’s ‘out’ event fires the first time the mouse moves into the canvas area even though the mouse was never over the red box. (This is most likely because the ‘over’ event was incorrectly fired so it probably thinks the mouse is over the red box to begin with.)

No longer occurs

> Right clicking causes the ‘over’ and ‘out’ events to fire when the right click content box shows up. Expectation: is that the context box is not part of the canvas and should not trigger events.

No longer occurs

**Comment by megakoresh** [comment]
@staff0rd thanks, but... How on earth do you find these things? Is there another documentation for this library that is actually complete?

**Comment by themoonrat** [comment]
Hi @megakoresh - The documentation for interaction can be found http://pixijs.download/release/docs/PIXI.interaction.html - but it seems that some of the docs aren't showing up, and some do need more added and clarification. You can click through to see the code on those pages, and thus see what is actually there, rather than just what the docs are outputting. Contributions to improve docs are always very welcome!

**Comment by staff0rd** [comment]
@megakoresh @themoonrat It seems that specific property was additionally marked `@private`.  I've opened PR #3471 to expose it.

**Comment by megakoresh** [comment]
@themoonrat It appears that the docs are autogenerated. I don't see how I could open a PR to help improve them.

**Comment by themoonrat** [comment]
Docs are generated from special comments written in the source code - documentation for which is at http://usejsdoc.org
So it's a PR just like any other :)

**Comment by hubletkoresh** [comment]
Ok I am not familiar with this. Maybe I can try to improve the docs for this library if I find things like that `moveWhenInside`.

**Comment by megakoresh** [comment]
Ok I am not familiar with this. Maybe I can try to improve the docs for this library if I find things like that moveWhenInside.

**Comment by staff0rd** [comment]
Last unsolved issue specific to OP has been moved to #3596.  Closing this, let me know if anyone thinks there's still any outstanding items.
